sips/sip-02x/sip-02x-stacks-pay.md

```diff
@@ -0,0 +1,343 @@
+
+**Stacks Pay** addresses this gap by introducing a payment request standard tailored specifically for the Stacks blockchain. The standard simplifies the process of sending and receiving payments, enhances interoperability among wallets and applications, and increases security by providing mechanisms to tie transactions back to payment requests.
+
+Stacks Pay does _not_ define smart contracts that may be used to facilitate specific functions, like subscriptions or NFT mints. Stacks Pay is strictly focused on defining the transaction payment information structure and encoding.
+
+## Operations
+
+Stacks Pay defines several standard operations that specify the type of payment action to be performed. Each operation type has specific required and optional parameters.
+
+### `support`
+
+**Function**: An open-ended, reusable request. Allows the payer to specify the amount and optionally the token, suitable for donations, tips, or gifts.
+
+**Required Parameters:**
+
+- operation: 'support'
+
+- recipient: A valid Stacks address.
+
+- If the token parameter is not 'STX', the functionName defaults to 'transfer' by the wallet.
+
+**Optional Parameters:**
+
+- token: Either 'STX' or a valid SIP-010 contract address. If not provided, defaults to 'STX'.
+
+- description: MAY be included to provide additional context for the payment request.
+
+- expiresAt: MAY be included; wallets MUST NOT process payment requests past the expiry time.
+
+- memo: MAY be included; MUST NOT include personal information.
+
+**Parameters to Ignore:**
+
+- amount, contractName, dueDate, functionName, invoiceNumber. Wallets MUST ignore these parameters if present.
+
+### `invoice`
+
+**Function:** Represents a payment request, such as an invoice for P2P payments or product purchases.
+
+**Required Parameters:**
+
+- operation: 'invoice'
+
+- recipient: A valid Stacks address.
+
+- token: MUST be either 'STX' or a valid SIP-010 contract address.
+
+- amount: The amount to be paid.
+
+**Optional Parameters:**
+
+- description: MAY be included to provide context for the payment request.
+
+- expiresAt: MAY be included; if included, wallets MUST NOT process payment requests past the expiry time.
+
+- invoiceNumber: MAY be included as an identifier.
+
+- dueDate: MAY be included.
+
+- memo: MAY be included. MUST NOT include personal information.
+
+**Parameters to Ignore:**
+
+contractName, functionName: Wallets MUST ignore these parameters if present.
+
+### `mint`
```

Comment:
I believe that enabling payments to smart contracts is within the realm of payments.  And malicious 'mint' operations can happen anywhere today.  At Boom we're putting default post conditions on every request.

Reply:
@dantrevino as SIP-009 doesn't specify a `mint` function in the spec this would mean that you can only mint Boom NFTs? As there is no spec for the params it wouldn't work with other platforms mint I guess

Reply:
Sorry no this is not intended to be a Boom thing.  The intention here was to make a generic name ... maybe to be more clear, for the mint function we make the `functionName` parameter required so that a wallet/app does not have to do the function lookup itself.